src/main/java/com/fasterxml/jackson/databind/util/StdDateFormat.java

```diff
@@ -131,12 +124,9 @@
      */
     protected Boolean _lenient;
     
-    protected transient DateFormat _formatRFC1123;
-    protected transient DateFormat _formatISO8601;
-    protected transient DateFormat _formatISO8601_z;
-    protected transient DateFormat _formatISO8601_noTz; // 2.8.10
-    protected transient DateFormat _formatPlain;
 
+    private transient Map<DateFormat, SimpleDateFormat> _clonedFormats = new HashMap<>();
```

Comment:
Maps are bit expensive, esp. when key is date format (depending on how `equals` and `hashCode` are defined).
What would work better would be `EnumMap` since that's basically just an array with index. So, create private `enum` with variants, pass that.

Although alternative could be to just always `clone()`/re-create instance for every call, and completely do away with recycling.

And yet third possibility would be to use `ThreadLocal`, with container object. That would probably work best as that would allow reuse across calls, without possibility of race conditions. The only (?) concern would be that of how to clear settings if they are changes. So maybe that doesn't work after all. :)